siwe.go

```diff
@@ -13,12 +13,67 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
-func InitMessage(domain, address, uri, version string, options map[string]interface{}) (*Message, error) {
-	validateURI, err := url.Parse(uri)
+func buildAuthority(uri *url.URL) string {
+	authority := uri.Host
+	if uri.User != nil {
+		authority = fmt.Sprintf("%s@%s", uri.User.String(), authority)
+	}
+	return authority
+}
+
+func validateDomain(domain *string) (bool, error) {
+	if isEmpty(domain) {
+		return false, &InvalidMessage{"`domain` must not be empty"}
+	}
+
+	validateDomain, err := url.Parse(fmt.Sprintf("https://%s", *domain))
+	if err != nil {
+		return false, &InvalidMessage{"Invalid format for field `domain`"}
+	}
+
+	authority := buildAuthority(validateDomain)
+	if authority != *domain {
+		return false, &InvalidMessage{"Invalid format for field `domain`"}
+	}
+
+	return true, nil
+}
+
+func validateURI(uri *string) (*url.URL, error) {
+	if isEmpty(uri) {
+		return nil, &InvalidMessage{"`uri` must not be empty"}
+	}
+
+	validateURI, err := url.Parse(*uri)
 	if err != nil {
 		return nil, &InvalidMessage{"Invalid format for field `uri`"}
 	}
 
+	return validateURI, nil
+}
+
+func InitMessage(domain, address, uri, version string, options map[string]interface{}) (*Message, error) {
```

Comment:
just wondering why issued at, nonce and expiration are in the options map since those are mandatory and should be always present?

Reply:
do we need to provide the version since it is always 1?

Reply:
can we provide inline doc for public / exported functions?

Reply:
+1 on @awoie suggestion on not making version string a required parameter here.